source/PlayerInfo.cpp

```diff
@@ -2579,3 +2621,46 @@ bool PlayerInfo::CanBeSaved() const
 {
 	return (!isDead && planet && system && !firstName.empty() && !lastName.empty());
 }
+
+
+
+// Called by conversation callbacks to destroy or capture the player's flagship.
+// Either method results in the death of the player.
+void PlayerInfo::Die(int response)
+{
+	if(GetPlanet() || !flagship)
+		Die(true);
+	else
+	{
+		Die();
+		auto it = find(ships.begin(), ships.end(), flagship);
+		if(response == Conversation::EXPLODE)
+			flagship->Destroy();
+		else
+		{
+			flagship->SetIsYours(false);
+			// If the flagship is boarding an NPC, the player's death results in its
+			// capture. Otherwise, consider the player to have died due to a mutiny.
+			// Change the flagship government to remove it from the EscortDisplay HUD.
+			if(boardingShip)
+			{
+				flagship->SetSwizzle(boardingShip->GetSwizzle());
+				flagship->WasCaptured(boardingShip);
+			}
+			else
+				flagship->SetGovernment(GameData::Governments().Get("Independent"));
+			
+			// Remove the now non-player ship from the player's fleet.
+			if(it != ships.end())
+				it = ships.erase(it);
+		}
+		
+		// Let the player's fleet pick a new flagship to follow, if possible.
+		for( ; it != ships.end(); ++it)
+			if((*it)->CanBeFlagship())
+				break;
+		const shared_ptr<Ship> newFlagship = it != ships.end() ? *it : nullptr;
+		for(const shared_ptr<Ship> &escort : ships)
+			escort->SetParent(escort != newFlagship ? newFlagship : nullptr);
```

Comment:
This seems unnecessary because the game is already over - it's no big deal if your escorts scatter instead of picking a new leader and departing together.

Reply:
I added this so that the player's fleet doesn't simply obey whoever killed the player, and follow that the player's "old" ship around (since it isn't destroyed, they won't scatter).